ksqldb-engine/src/main/java/io/confluent/ksql/internal/TelemetryMetricsReporter.java

```diff
@@ -0,0 +1,123 @@
+package io.confluent.ksql.internal;
+
+import com.google.common.collect.ImmutableMap;
+import io.confluent.telemetry.MetricKey;
+import io.confluent.telemetry.MetricsUtils;
+import io.confluent.telemetry.metrics.SinglePointMetric;
+import io.confluent.telemetry.provider.KsqlProvider;
+import io.confluent.telemetry.provider.ProviderRegistry;
+import io.confluent.telemetry.reporter.TelemetryReporter;
+import org.apache.kafka.common.metrics.KafkaMetricsContext;
+import org.apache.kafka.common.metrics.MetricsContext;
+import org.apache.kafka.common.utils.AppInfoParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TelemetryMetricsReporter implements MetricsReporter {
+
+    public static final String RESOURCE_LABEL_PREFIX = "resource.";
+    public static final String RESOURCE_LABEL_TYPE = RESOURCE_LABEL_PREFIX + "type";
+    public static final String RESOURCE_LABEL_VERSION = RESOURCE_LABEL_PREFIX + "version";
+    public static final String RESOURCE_LABEL_COMMIT_ID = RESOURCE_LABEL_PREFIX + "commit.id";
+    public static final String RESOURCE_CLUSTER_ID = RESOURCE_LABEL_PREFIX + "cluster.id";
+
+    private final Logger logger = LoggerFactory.getLogger(TelemetryMetricsReporter.class);
+    private final TelemetryReporter reporter;
+    private final Map<String, String> metricTags = new HashMap<>();
+    // ToDo: How is the cluster ID passed to this class
+    private final String clusterId = "";
```

Comment:
Where do we get tags like cluster ID from?

Reply:
@rodesai will know the structure better than I do but he and I talked a bit about this yesterday, cluster ID and node ID should come from the spec or scheduler call. In the HC we got them from a tag list in the config, they might be there for ksql as well, but I'd assume there's an easier way to pass it in than parsing a tag map 😀